objectstorage/transfer/file_uploader.go

```diff
@@ -15,13 +15,13 @@ import (
 // FileUploader is an interface to upload a file
 type FileUploader interface {
 	// split file into multiple parts and uploads them to blob storage, then merge
-	UploadFileMultiparts(ctx context.Context, request UploadFileRequest) (response UploadResponse, err error)
+	UploadFileMultiparts(ctx context.Context, request UploadFileRequest, callBack objectstorage.UploadCallBack) (response UploadResponse, err error)
```

Comment:
these public interface changes are breaking existing customers. Can you put callback inside request object and make it optional?

Reply:
ok, done

Reply:
sure, done